Core/Graphics/Source/DeviceImpl.cpp

```diff
@@ -390,8 +398,7 @@ namespace Babylon::Graphics
         uint32_t frameNumber{bgfx::frame()};
 
         // Process read texture requests.
-        assert(m_readTextureRequests.empty() || m_readTextureRequests.front().first >= frameNumber);
-        while (!m_readTextureRequests.empty() && m_readTextureRequests.front().first == frameNumber)
+        while (!m_readTextureRequests.empty() && m_readTextureRequests.front().first <= frameNumber)
```

Comment:
Are the readPixels pending promises in this situation still valid? Or should they just be error'd out like you do above?

Reply:
As far as I can tell the results are still valid.  I'll set up a test to explicitly delay the sending of results to double check.

Reply:
The results are still valid up to at least 1000 frames later.  I don't think we'll ever hit a scenario where this goes longer than 1-2 frames over.